packages/agreement-process/src/model/domain/agreement-validators.ts

```diff
@@ -155,23 +157,43 @@ const assertRequesterIsProducer = (
 
 export const assertRequesterCanActAsConsumerOrProducer = (
   agreement: Agreement,
-  authData: UIAuthData | M2MAdminAuthData,
-  activeDelegations: ActiveDelegations
+  delegationId: DelegationId | undefined,
+  activeDelegations: ActiveDelegations,
+  authData: UIAuthData | M2MAdminAuthData
 ): void => {
-  try {
-    assertRequesterCanActAsConsumer(
-      agreement.consumerId,
-      agreement.eserviceId,
-      authData,
-      activeDelegations.consumerDelegation
-    );
-  } catch {
-    try {
-      assertRequesterCanActAsProducer(
+  if (delegationId) {
+    if (delegationId === activeDelegations.producerDelegation?.id) {
+      return assertRequesterIsDelegateProducer(
         agreement,
         authData,
         activeDelegations.producerDelegation
       );
+    } else if (delegationId === activeDelegations.consumerDelegation?.id) {
+      return assertRequesterIsDelegateConsumer(
+        agreement,
+        authData,
+        activeDelegations.consumerDelegation
+      );
+    } else {
+      throw tenantNotAllowed(authData.organizationId);
```

Comment:
Also here I would be more specific, the tenant could be allowed but the issue may be in the delegation ID they pass
```suggestion
      throw unauthorizedError("Tenant cannot perform operation as delegate for the specified delegation ID");
```